test/e2e/25_max_history.bats

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+
+function setup() {
+  # Load libraries in setup() to access BATS_* variables 
+  load lib/env
+  load lib/install
+  load lib/poll
+
+  kubectl create namespace "$E2E_NAMESPACE"
+  install_git_srv
+  install_tiller
+  install_helm_operator_with_helm
+  kubectl create namespace "$DEMO_NAMESPACE"
+}
+
+@test "When max history is not set on release the most recent 10 revisions are kept" {
+  if [ "$HELM_VERSION" != "v3" ]; then
+    skip
+  fi
+
+  # Apply the HelmRelease
+  kubectl apply -f "$FIXTURES_DIR/releases/helm-repository.yaml" >&3
+  poll_until_equals 'podinfo-helm-repository HelmRelease' 'deployed' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.releaseStatus' --no-headers"
+
+  for i in {1..15}
+  do
+    # Apply a patch to initiate an upgrade
+    kubectl patch -f "$FIXTURES_DIR/releases/helm-repository.yaml" --type='json' -p="[{\"op\": \"replace\", \"path\": \"/spec/values/someField\", \"value\":\"$i\"}]" >&3
+    poll_until_equals 'podinfo-helm-repository HelmRelease' "$(($i+1))" "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.observedGeneration' --no-headers"
+  done
+
+  # Check count of revisions is <= 10
+  count=$(kubectl exec -n "$E2E_NAMESPACE" deploy/helm-operator -- helm3 history podinfo-helm-repository --namespace "$DEMO_NAMESPACE" --skip-headers | tail -n +2 | wc -l)
```

Comment:
Could not find a way to not output headers like you can with kubectl in table mode so skipping the first line.